lib/apiservers/engine/backends/archive.go

```diff
@@ -644,3 +687,90 @@ func (rm *ArchiveStreamReaderMap) Close() {
 
 	rm.prefixTrie.Visit(closeStream)
 }
+
+// try to fake the statpath for path that targets the mountpoint or along the mountpoint
+func tryFakeStatPath(mounts []types.MountPoint, target string) (*types.ContainerPathStat, bool) {
+	isMountPathTarget := false
+
+	for _, mount := range mounts {
+		if strings.HasPrefix(mount.Destination, target) {
+			isMountPathTarget = true
+		}
+	}
+
+	// check to see if the path is a mount point, if so, return fake path
+	if isMountPathTarget {
+		return &types.ContainerPathStat{
+			Name:       filepath.Base(target),
+			Size:       int64(4096),
+			Mode:       os.ModeDir,
+			Mtime:      time.Now(),
+			LinkTarget: ""}, true
+	}
+
+	return nil, false
+}
+
+// use mountpoints to strip the target to a relative path
+func resolvePathWithMountPoints(op trace.Operation, mounts []types.MountPoint, path string) *ArchiveReader {
+	var primaryTarget *ArchiveReader
+
+	readerMap := NewArchiveStreamReaderMap(op, mounts, path)
+	nodes, _ := readerMap.FindArchiveReaders(path)
+
+	for _, node := range nodes {
+		if strings.HasPrefix(path, node.mountPoint.Destination) &&
+			(primaryTarget == nil || len(node.mountPoint.Destination) > len(primaryTarget.mountPoint.Destination)) {
+			primaryTarget = node
+		}
+	}
+
+	return primaryTarget
+
+}
+
+func (rm *ArchiveStreamReaderMap) buildFilterSpec(containerSourcePath string, nodes []*ArchiveReader, startingNode *ArchiveReader) error {
+
+	mounts, foundNodes, err := rm.buildMountsAndNodes(startingNode.mountPoint.Destination, startingNode)
+	if err != nil {
+		return err
+	}
+
+	for _, node := range foundNodes {
+		vicarchive.AddMountInclusionsExclusions(node.mountPoint.Destination, &node.filterSpec, mounts, containerSourcePath)
+	}
+	return nil
+}
+
+// This function will return the node pointers from the prefix tree as well as all mounts involved in the operation
+func (rm *ArchiveStreamReaderMap) buildMountsAndNodes(path string, node *ArchiveReader) ([]string, []*ArchiveReader, error) {
+
+	// We can modify this to make proper exclusions in the future. For now,
```

Comment:
It would be nice to prefix this comment block with a `NOTE(author):` so it's easier to address in the future.